### PR TITLE
Update to 9.0.0, add plugin version 2 and 3, update tracer and tiny examples with register output.

### LINKED ISSUE #15: qemu plugins made with this don't work with Qemu 9.0.0 or later
The minimum supported plugin version was bumped to 2 in Qemu 9.0.0, and this library makes plugins with v1, and so they do not work in the newest Qemu. You get output like:

```
qemu-system-x86_64: Could not load plugin plugin.dll: plugin requires API version 1, but this QEMU supports only a minimum version of 2
```

### PR DESCRIPTION
**Pull request by novafacing**
Should resolve #15. This PR adds support for new API versions 2 (for 9.0.0) and 3 (for >9.0.0) and their associated functionality. The demo plugins are also updated with register reading (the most exciting new feature).

### CODE REVIEW

**Comment by novafacing** [comment]
@tocklime if you have some time, maybe you can review this. I didn't have a chance to test on Windows, but everything looks good on Linux.

**Comment by novafacing** [comment]
A couple odds and ends not in here yet:

- Update documentation about the plugin API version now being feature-flag-controlled (`plugin-api-vX`)
- Add a convenience function for compiling a plugin thrice for each version? Maybe not necessary.

File: qemu-plugin-sys/generate-bindings.rs [hunk 1]
```diff
@@ -18,20 +22,33 @@ use bindgen::{
...
+fn qemu_src_url_v1() -> String {
+    format!("{}qemu-{}.tar.xz", QEMU_SRC_URL_BASE, QEMU_VERSION_V1)
+}
+
+fn qemu_src_url_v2() -> String {
+    format!("{}qemu-{}.tar.xz", QEMU_SRC_URL_BASE, QEMU_VERSION_V2)
+}
+
+fn qemu_src_url_v3() -> String {
+    format!("{}/archive/4abc892362f8282450f18c4e45c5b0534461d01e.zip", QEMU_GITHUB_URL_BASE)
```
Comment (tocklime) [inline qemu-plugin-sys/generate-bindings.rs:51]:
```suggestion
    format!("{}/archive/{}.zip", QEMU_GITHUB_URL_BASE, QEMU_VERSION_V3)
```
?

File: qemu-plugin-sys/generate-bindings.rs [hunk 2]
```diff
@@ -74,21 +91,40 @@ fn extract_txz(archive: &Path, destination: &Path) -> Result<()> {
     Ok(())
 }
 
-#[cfg(windows)]
-fn generate_windows_delaylink_library(qemu_plugin_symbols: &Path, out_dir: &Path) -> Result<()> {
-    let def_file = out_dir.join("qemu_plugin_api.def");
+/// Extract a zip frile at a path to a destination
```
Comment (tocklime) [inline qemu-plugin-sys/generate-bindings.rs:94]:
```suggestion
/// Extract a zip file at a path to a destination
```

File: qemu-plugin-sys/generate-bindings.rs [hunk 3]
```diff
@@ -74,21 +91,40 @@ fn extract_txz(archive: &Path, destination: &Path) -> Result<()> {
...
     let all_commands = read_to_string(qemu_plugin_symbols)?;
     let all_commands = all_commands.replace(|x| "{};".contains(x), "");
-    write(&def_file, format!("EXPORTS\n{all_commands}"))?;
+    write(destination, format!("EXPORTS\n{all_commands}"))?;
 
     Ok(())
 }
 
 fn generate_bindings(qemu_plugin_header: &Path, destination: &Path) -> Result<()> {
+    // let glib_20 = probe_library("glib-2.0").map_err(|e| anyhow!(e))?;
```
Comment (tocklime) [inline qemu-plugin-sys/generate-bindings.rs:111]:
```suggestion
```

**Review by tocklime — APPROVED** [review]
This is awesome, thanks. Have built it in msys. The tiny plugin works, and is giving me veeeeery verbose register output. Fun times. tracer doesn't compile on windows (use of UnixStream), but that's not new.

The plugin I'm writing internally compiles against this version fine (though I've not done the register stuff yet).

I spotted a couple of typos/dead code things.

**Pull request merged**